docs/DeveloperGuide.md

```diff
@@ -193,6 +193,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+### User Login Command
+
+**Overview:**
+
+The `login` command is used to log in to the user's account.
+Once logged in, the user will have access to all the commands available.
+
+The format for the `login` command can be found [here](UserGuide.md#login).
+
+**Feature details:**
+
+1. The user specifies the `Username` and `Password` in the `login` command.
+2. If any of the fields is not provided, an error message with the correct command usage will be shown.
+3. If invalid command parameters are provided, an error message with the correct parameter format will be shown.
+4. The `User` is then cross-referenced with the stored user in `Model` to check if the credentials match.
+If incorrect credentials are provided, an error message regarding wrong credentials will be shown.
+5. If the user is currently logged in, an error message will be shown.
+6. If all the previous steps are completed without exceptions, the user will be logged in and the 
+`isLoggedIn` status in `Model` will be updated.
+
+The following activity diagram shows the logic of a user logging in:
+
+<puml src="diagrams/UserLoginActivityDiagram.puml" alt="UserLoginActivityDiagram" />
+
+The sequence of the `login` command is as follows:
+
+1. Upon launching the application, the `ModelManager` will be initialized with 
+the `User` constructed with details from the authentication.json file.
+2. The user inputs the `login` command with the username and password.
+3. The `userLoginCommandParser` checks whether all the required fields are present.
+If all fields are present, it creates a new `userLoginCommand`.
+4. The `userLoginCommand` checks whether the user is currently logged in by calling `Model#getUserLoginStatus()`. 
+5. The `userLoginCommand` then checks if the user credentials match the stored user by calling `Model#userMatches()`.
```

Comment:
In the line above it says it check if the user credentials is correct before checking if the user is logged in, but here it is the other way round

Reply:
Updated the section prior to this